cmd/sovereignnode/chainSimulator/tests/processBlock/cleanupHeaders_test.go

```diff
@@ -0,0 +1,82 @@
+package processBlock
+
+import (
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	sovereignChainSimulator "github.com/multiversx/mx-chain-go/sovereignnode/chainSimulator"
+)
+
+func TestSovereignChainSimulator_BlockTrackerPoolsCleanup(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	cs, err := sovereignChainSimulator.NewSovereignChainSimulator(sovereignChainSimulator.ArgsSovereignChainSimulator{
+		SovereignConfigPath: sovereignConfigPath,
+		ArgsChainSimulator: &chainSimulator.ArgsChainSimulator{
+			BypassTxSignatureCheck: false,
+			TempDir:                t.TempDir(),
+			PathToInitialConfig:    defaultPathToInitialConfig,
+			GenesisTimestamp:       time.Now().Unix(),
+			RoundDurationInMillis:  uint64(6000),
+			RoundsPerEpoch:         core.OptionalUint64{},
+			ApiInterface:           api.NewNoApiInterface(),
+			MinNodesPerShard:       2,
+			ConsensusGroupSize:     2,
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	logger.SetLogLevel("*:DEBUG")
+	nodeHandler := cs.GetNodeHandler(core.SovereignChainShardId)
+
+	headerNonce := uint64(9999999)
+	prevHeader := createHeaderV2(headerNonce, generateRandomHash(), generateRandomHash())
+
+	for round := 1; round <= 10; round++ {
+		err = cs.GenerateBlocks(1)
+		require.Nil(t, err)
+
+		selfNotarizedHeaders, _, err := nodeHandler.GetProcessComponents().BlockTracker().GetSelfNotarizedHeader(core.SovereignChainShardId, 0)
+		require.Nil(t, err)
+		require.NotNil(t, selfNotarizedHeaders)
+		require.Equal(t, uint64(round)-1, selfNotarizedHeaders.GetNonce()) // in round X, header with nonce X-1 is notarized
+
+		crossTrackedHeaders, _ := nodeHandler.GetProcessComponents().BlockTracker().GetTrackedHeaders(core.MainChainShardId)
+		require.Nil(t, err)
+		if round == 1 {
+			require.Nil(t, crossTrackedHeaders)
+		} else {
+			require.NotNil(t, crossTrackedHeaders)
+			require.Equal(t, headerNonce, crossTrackedHeaders[len(crossTrackedHeaders)-1].GetNonce())
+			require.LessOrEqual(t, len(crossTrackedHeaders), 4)
```

Comment:
Let's perhaps add a comment here why is 4. From what I remember I think it's because we have 2 dummy headers that remain "stuck" there

Reply:
3 are the ones notarzied + 1 tracked